functional/transformations.py

```diff
@@ -611,7 +611,7 @@ def grouped_impl(wrap, size, sequence):
     iterator = iter(sequence)
     while True:
         batch = islice(iterator, size)
-        yield chain((wrap(next(batch)),), batch)
+        yield list(chain((wrap(next(batch)),), batch))
```

Comment:
Can you add a unit test to make sure the behavior described in the issue is actually fixed?

Reply:
The test `test_grouped()` was failing before this change. Do you want me to add another test?

Reply:
The test seems to have been passing, but not performing a check to catch this. Looks like they test that for iterator equality, but don't actually test that the elements are lists which should be easy to fix.